azavea/cac-tripplanner
# Various UI improvements

## Overview

A collection of UI improvements, including:

#### 2017 enhancements
- App icon assets for mobile web app / PWA. Connects #865 
- Style current location marker. Connects #867 
- Show geolocation control in map view. Resolves #835 
- Move place page links up. Resolves #840 
- Visually distinguish blog post card on home page. Resolves #841 

#### iOS layout bugs
- Remove gap from bottom of iPhone directions view. Fixes #878
- Stop iPad map sidebar from wrapping. Fixes #877
- Stop footer from obscuring content on info pages (iOS9). Fixes #871 


### Demo

#### App icons

Android 7
![image](https://user-images.githubusercontent.com/128699/29087021-80822a26-7c42-11e7-97e0-e2e688f719b3.png)

iOS 10
![image](https://user-images.githubusercontent.com/128699/29087123-d92ec198-7c42-11e7-856a-01945e825038.png)

#### Current location marker
![image](https://user-images.githubusercontent.com/128699/29087276-5dcc8688-7c43-11e7-8842-02a6fb3c9e25.png)

#### Geolocation control in map view
![image](https://user-images.githubusercontent.com/128699/29087169-0c039472-7c43-11e7-8e02-7b6e1937d262.png)

#### Move place page links up
![image](https://user-images.githubusercontent.com/128699/29087297-73fcbf22-7c43-11e7-9495-0d1d4f9c58a9.png)

#### Visually distinguish blog post card on home page
![image](https://user-images.githubusercontent.com/128699/29087361-a722eba6-7c43-11e7-9aab-8f76e4dffda6.png)


### Notes

#### Current location marker

Some browsers require the page be served over https to use the geolocation API. So we probably won't be able to test the current location marker styling across all browsers until this is on staging.

#### Geolocation control in map view

This required moving the "reverse route" button down into the destination field. It's not optimal but it's a tidy compromise that works.


## Testing Instructions

* App icon assets
  * Load the site in Android (v7) Chrome and iOS (10) Mobile Safari (browserstack or simulator should be fine). Add it to home screen. Home screen should like above.
* Current location marker
  * In Chrome desktop, load site in narrow window (width is used as proxy for mobile). Go to map view. Current location should appear on map, as above.
* Geolocation control in map view
  * Go to map view. Geolocation button should appear inside origin field and should be operational.
* Move place page links up
  * Go to a place page ("More info"). Website and Directions buttons should appear as above, and be functional.
* Visually distinguish blog post card on home page
  * Go to home page, set viewport to mobile width. Blog post card should appear as above.
